structure-macro-impl/src/lib.rs

```diff
@@ -2,9 +2,25 @@
 
 extern crate proc_macro;
 
+#[cfg(not(feature = "std"))]
+extern crate core;
+#[cfg(not(feature = "std"))]
+extern crate alloc;
+
+#[cfg(not(feature = "std"))]
+3use alloc::string::String;
+#[cfg(not(feature = "std"))]
```

Comment:
You don't need to do this. Anything that is available in `core` can be imported from there, wheteher `std` is used or not.